include/votca/csg/atomcgconverter.h

```diff
@@ -0,0 +1,195 @@
+  AtomCGConverter(){};
+
+  /**
+   * @brief Loads the .xml file describing the cg molecule
+   *
+   * Will load the information from the cg.xml file and create a stencil from
+   * the input. For each cg molecule defined.
+   *
+   * @param[in] filename should be an .xml file
+   */
+  void LoadMoleculeStencil(std::string filename);
+
+  /**
+   * @brief Provided the atomic type of the molecule returns the cg type of the
+   * molecule
+   *
+   * @param[in] atomistic_molecule_type
+   *
+   * @return cg molecule type as a string
+   */
+  const std::string &getCGMoleculeType(
+      std::string atomistic_molecule_type) const;
+
+  /**
+   * @brief Provided the cg type of the molecule returns the atomic type of the
+   * molecule
+   *
+   * @param cg_molecule_type
+   *
+   * @return atomistic molecule type as a string
+   */
+  const std::string &getAtomisticMoleculeType(
+      std::string cg_molecule_type) const;
+
+  /**
+   * @brief Converts a atomic topology to a coarsegrained topology
+   *
+   * @param atomic_top_in
+   * @param cg_top_out
+   */
+  void Convert(CSG_Topology &atomic_top_in, CSG_Topology &cg_top_out);
+
+  /**
+   * @brief Updates the cg topology object using the atomistic topology
+   *
+   * The properties updated include the time step, bead positions, forces and
+   * velocities etc...
+   *
+   * @param atomic_top_in
+   * @param cg_top_out
+   */
+  void Map(CSG_Topology &atomic_top_in, CSG_Topology &cg_top_out);
+
+  std::vector<std::string> getAtomicBeadNamesOfCGBead(
+      std::string cg_molecule_type, std::string cg_bead_type);
+
+  bool AtomisticMoleculeTypeExist(std::string atomistic_molecule_type);
+
+ private:
+  std::unordered_set<std::string> file_names_;
+
+  std::map<int, std::map<int, std::vector<std::pair<std::string, int>>>>
+      cgmolid_cgbeadid_atomicbeadnames_and_ids_;
+
+  std::unordered_set<std::string> atomic_mol_types_to_ignore_;
+  /**
+   * @brief Atomic and cg bimap stores the relationship between the two types
+   * of molecules
+   */
+  boost::bimap<std::string, std::string> atomic_and_cg_molecule_types_;
+
+  /**
+   * @brief Stores the stencil of the cg molecule
+   *
+   * string is the cg_molecule_type
+   */
+  std::unordered_map<std::string, CGMoleculeStencil> cg_molecule_and_stencil_;
+
+  // First string atomic_molecule_type
+  // Second string cg_molecule_type
+  // The molecule map
+  std::unordered_map<std::string,
+                     std::unordered_map<std::string, AtomToCGMoleculeMapper>>
+      mol_names_and_maps_;
+
+  /**
+   * @brief Maps a atomistic molecule to a cg molecule and adds it to the cg
+   * topology
+   *
+   * Note that the cg and atomistic descriptions should have the same molecule
+   * ids, this means that the molecule ids will remain consistent between the
+   * coarse grained and atomistic descriptions.
+   *
+   * @param[in] atomistic_molecule
+   * @param[in,out] cg_top_out
+   */
+  void ConvertAtomisticMoleculeToCGAndAddToCGTopology_(
```

Comment:
waht is a CGT topology why not use a return type?

Reply:
atom top should probably be const.

Should this method not be two separate ones?